lib/auth/init.go

```diff
@@ -637,6 +639,22 @@ func initializeAuthorities(ctx context.Context, asrv *Server, cfg *InitConfig) e
 		return trace.Wrap(err)
 	}
 
+	// Collect CAs from integrations to avoid deleting them.
+	igs, err := clientutils.ListAllResources(ctx, asrv.Services.ListIntegrations)
```

Comment:
How many resources do we expect this to return? 

There's not really a need to buffer them all in memory before you start collecting the keys. If there are going to be many pages it will be more memory efficient to iterate a page at a time and collect the keys per-page.

Reply:
i would say a handful at most but agreed we shouldn't encourage holding everything in memory. changed it to an iterator.